node/core/approval-voting/src/criteria.rs

```diff
@@ -483,50 +619,100 @@ pub(crate) fn check_assignment_cert(
 	let public = schnorrkel::PublicKey::from_bytes(validator_public.as_slice())
 		.map_err(|_| InvalidAssignment(Reason::InvalidAssignmentKey))?;
 
-	if claimed_core_index.0 >= config.n_cores {
-		return Err(InvalidAssignment(Reason::CoreIndexOutOfBounds))
+	// Check that we have all backing groups for claimed cores.
+	if claimed_core_indices.is_empty() && claimed_core_indices.len() != backing_groups.len() {
+		return Err(InvalidAssignment(Reason::InvalidArguments))
```

Comment:
not sure I understand the first check, should it be `len != backing_groups.len()`?
why `empty` is required and why `&&`?